.github/workflows/upload_app_store.yml

```diff
@@ -0,0 +1,66 @@
+name: Upload IPA to App Store
+on: workflow_dispatch
```

Comment:
This makes sense, to call it explicitly, but then you're replicating part of the functionality of ios-test.yaml here. So have you considered these other options?
1. Set this a a job that triggers only if build works *and* event is workflow dispatch (or other you might think of, such as tagging with a certain pattern).
2. Reuse steps: [this is possible since a few months ago]8https://docs.github.com/en/actions/using-workflows/reusing-workflows), and it would imply spinning off the common part from ios-test.yaml and this one.
3. Caching builds, which is probably only worth the while if computing the hash for the cache is significantly faster than compiling (and compiling is fast). 

At any rate, considering the existing workflow (which is always in flow) might help, if you must.

Reply:
I was thinking of point number 2, but I thought to have this in a simpler configuration initially. I will look into reusing the iOS-test.yml, which looks to be the most appropriate solution to me.

Reply:
@JJ I've looked more on how reusing workflows/steps might work with GitHub Actions. From what I have understood I can only chain the two workflows by creating a job for the chained workflow. This means that job workspace will be recreated and a new checkout is made. The artefacts could be shared between jobs, so we don't rebuild again.
But, from iOS-test.yaml I am actually reusing only `make test`. The rest is the workflow setup.

I am in favour of keeping it as is for this workflow. WDYT?

Reply:
Well, it's everything up to and including make test, essentially, everything that's now in ios-test.yaml It's true that since they are in different jobs you would need to pass artifacts from one to the another, or replicate checkout and build, at least. Mid term solution is that we really need to take care of build artifacts so that we need to re-use them among workflows. But it's true that the short term solution would include replicating a few steps. So you're right, just keep as it is.